### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by Bhekisile**
## Changes

- Added validations for the user model `(name` and` postsCounter`).
- Added validations for the post model (`title`,` commentsCounter`, and` likesCounter`).
- Created unit tests for models' methods and validations

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      last_post.comments.create(post: last_post, user: user1, text: 'Comment three!')
+      last_post.comments.create(post: last_post, user: user1, text: 'Comment four!')
+      last_post.comments.create(post: last_post, user: user1, text: 'Comment five!')
+      last_post.comments.create(post: last_post, user: user1, text: 'Comment six!')
+
+      recent_comment = last_post.recent_comments
+      expect(recent_comment.size).to eq(5)
+    end
+  end
+end
```
Comment (Munsa1) [inline spec/models/post_spec.rb]: - [ ] As required, we should add tests for all custom methods, kindly add a test scenario for the `update_user_posts_counter` method found in the Post model

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Bhekisile, 
required changes

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Added Validations :heavy_check_mark: 
Passing Tests Cases :heavy_check_mark: 
Descriptive Pr :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 1 above)
Comment (Bhekisile) [reply spec/models/post_spec.rb]: Fixed, thank you,

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @Bhekisile, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**